test/hotspot/jtreg/gc/TestJNICriticalStressTest.java

```diff
@@ -0,0 +1,238 @@
+                }
+                System.out.println("SYSTEM_GC BEFORE");
+                wb.fullGC();
+                fullGcCounts += 1;
+                System.out.println("SYSTEM_GC AFTER");
+
+                long gcCountAfter = collector.getCollectionCount();
+                Asserts.assertLessThanOrEqual(gcCountBefore + fullGcCounts, gcCountAfter, "Triggered more Full GCs than expected");
+            }
+        }
+    }
+
+    static public Map<Integer,String> largeMap;
+
+    public static void main(String... args) throws Exception {
+        if (args.length < 4) {
+            println("usage: JNICriticalStressTest3 <duration sec> <alloc threads> <jni critical threads>");
+            exit(-1);
+        }
+
+        long durationSec = Long.parseLong(args[0]);
+        int allocThreadNum = Integer.parseInt(args[1]);
+        int jniCriticalThreadNum = Integer.parseInt(args[2]);
+
+        StringBuilder OldGCName = new StringBuilder();
+        switch (args[3]) {
+            case "G1":
+                OldGCName.append("G1 Old Generation");
+                break;
+            case "Parallel":
+                OldGCName.append("PS MarkSweep");
+                break;
+            case "Serial":
+                OldGCName.append("MarkSweepCompact");
+                break;
+            default:
+                throw new RuntimeException("Unsupported GC selected");
+        }
+
+        List<GarbageCollectorMXBean> collectors = ManagementFactory.getGarbageCollectorMXBeans();
+
+
+        for (int i = 0; i < collectors.size(); i++) {
+            println(collectors.get(i).getName());
+            if (collectors.get(i).getName().contains(OldGCName.toString())) {
+                collector = collectors.get(i);
+                break;
+            }
+        }
+
+        if (collector == null) {
+            throw new RuntimeException(OldGCName.toString() + " not found");
+        }
+
+        println("Running for " + durationSec + " secs");
+
+        largeMap = populateMap(LARGE_MAP_SIZE);
+
+        println("Starting " + allocThreadNum + " allocating threads");
+        for (int i = 0; i < allocThreadNum; i += 1) {
+            new Thread(new AllocatingWorker()).start();
+        }
+
+        println("Starting " + jniCriticalThreadNum + " jni critical threads");
+        for (int i = 0; i < jniCriticalThreadNum; i += 1) {
+            new Thread(new JNICriticalWorker()).start();
+        }
+
+        gcCountBefore = collector.getCollectionCount();
+
+        new Thread(new SystemGCWorker()).start();
+
+        long durationMS = (long) (1000 * durationSec);
+        long start = System.currentTimeMillis();
+        long now = start;
+        long soFar = now - start;
+        while (soFar < durationMS) {
+            try {
+                Thread.sleep(durationMS - soFar);
+            } catch (Exception e) {
+            }
+            now = System.currentTimeMillis();
+            soFar = now - start;
+        }
```

Comment:
I believe it's unexpected for the current thread to be interrupted; if so, it should log the exception and exit peacefully, as there's no enough info here to properly process it.

```java
try {
    Thread.sleep(durationSec * 1000);
} catch (InterruptedException e) {
    e.printStackTrace();
    return;
}
```